javascript/ql/src/semmle/javascript/Expr.qll

```diff
@@ -245,6 +245,21 @@ class Expr extends @expr, ExprOrStmt, ExprOrType, AST::ValueNode {
       ctx.(ConditionalExpr).inNullSensitiveContext()
     )
   }
+
+  /**
+   * Gets the data-flow node where exceptional data-flow will flow if this expression
+   * causes an exception to be thrown.
+   */
+  DataFlow::Node getThrowsToNode() {
+    if exists(this.getEnclosingStmt().getEnclosingTryStmt())
```

Comment:
Don't you also have to check whether that `try` statement has a `catch` clause? It might just have a `finally`.

Reply:
You are right. 
I refactored the default taint-step for exceptional flow to use that method, so I'm surprised that no tests failed.

Reply:
That suggests we might benefit from a few more tests.

Reply:
I added a test that tests that feature in the `js/xss` query.